sound/soc/sof/intel/hda.c

```diff
@@ -1427,3 +1461,4 @@ MODULE_IMPORT_NS(SOUNDWIRE_INTEL_INIT);
 MODULE_IMPORT_NS(SOUNDWIRE_INTEL);
 MODULE_IMPORT_NS(SND_SOC_SOF_HDA_MLINK);
 MODULE_IMPORT_NS(SND_SOC_SOF_INTEL_HDA_COMMON);
+MODULE_IMPORT_NS(SND_SOC_INTEL_SOF_SSP_COMMON);
```

Comment:
This looks like a circular dependency to me. It's problematic if the PCI parent module depends on the board level stuff.

Reply:
How about moving the code of ssp-common to SND_SOC_ACPI_INTEL_MATCH module? They are similar in many ways. Both modules handle codec/amp detection via ACPI. Both are designed for machine driver enumeration.

Reply:
That sounds ok at a high-level, we just need to make sure the dependencies are well managed and that this wouldn't break CI with the sof-test/tools/kmod scripts doing things that no longer work.

Reply:
Another PR #4852 is submitted